src/drivers/driver.ts

```diff
@@ -1231,47 +1359,75 @@ export abstract class CMakeDriver implements vscode.Disposable {
     }
   }
 
-  async getCMakeBuildCommand(target: string): Promise<proc.BuildCommand|null> {
-    const gen = this.generatorName;
-    target = this.correctAllTargetName(target);
+  async getCMakeBuildCommand(target?: string): Promise<proc.BuildCommand|null> {
+    if (this.useCMakePresets) {
```

Comment:
I'm having a problem here because this is always returning false for me even though cmake.useCMakePresets is set to true in my settings.  It doesn't look like `this._useCMakePresets` is ever updated once the driver is created.

Reply:
The driver is disposed when switching... Looks like I need to reset it to null as well... This is great finding and might be the reason why my test was failing yesterday.